lib/Routing/Dynamic/DataObjectRouteHandler.php

```diff
@@ -114,6 +123,17 @@ private function buildRouteForFromSlug(DataObject\Data\UrlSlug $slug, DataObject
         $route->setDefault('object', $object);
         $route->setDefault('urlSlug', $slug);
 
+        trigger_deprecation(
```

Comment:
I'd say the deprecation should not be triggered if the value is the one that is not deprecated. I.e. it should only be triggered if the value is `false`. Otherwise, you can't fix the deprecation in your code prior to updating to the version where the behavior changes.

Reply:
This is the wrong place anyway, that needs to be done in ElementListener as mentioned before. 
Here should be the right place: 
https://github.com/pimcore/pimcore/blob/e37d843d43299593391baa3d24e46cd442cf82ec/bundles/CoreBundle/EventListener/Frontend/ElementListener.php#L96-L95

Something like the following: 
```php
                if($request->get('object') && $request->get('urlSlug')) {
                    trigger_deprecation(...);
                }
```